### PR TITLE
[MRRTF-215] Small additions to ElecMap

### PR DESCRIPTION
**Pull request by aphecetche**
Adding functions to convert solar ids to/from indices, and to convert a detection element id into a detection element index (0..155)

### CODE REVIEW

File: Detectors/MUON/MCH/Raw/ElecMap/include/MCHRawElecMap/Mapper.h [hunk 1]
```diff
@@ -125,6 +125,17 @@ extern std::array<int, 13> deIdsOfCH10L;
 template <typename T>
 std::vector<std::string> solar2FeeLinkConsistencyCheck();
 
+/** deIndex returns an index (in the range 0..155) corresponding the deId. */
+std::optional<int> deId2DeIndex(int deId);
```
Comment (pillot) [inline Detectors/MUON/MCH/Raw/ElecMap/include/MCHRawElecMap/Mapper.h]: Maybe a detail, but I think this function would be better placed in MCHConstants/DetectionElements.h, where the array of DE, making the reverse connexion deIdx --> deId, is defined.

File: Detectors/MUON/MCH/Raw/ElecMap/src/ElectronicMapperDummy.cxx [hunk 2]
```diff
@@ -171,4 +171,10 @@ std::set<DsDetId> getDualSampasPerFeeId<ElectronicMapperDummy>(uint16_t feeId)
   return impl::getDualSampasPerFeeId<ElectronicMapperDummy>(feeId);
 }
 
+template <>
+std::optional<uint16_t> solarIndex2Id<ElectronicMapperDummy>(uint16_t solarIndex)
+{
+  return impl::solarIndex2Id<ElectronicMapperDummy>(solarIndex);
+}
+
```
Comment (pillot) [inline Detectors/MUON/MCH/Raw/ElecMap/src/ElectronicMapperDummy.cxx:179]: not sure it is useful, but the reverse operation it not defined for this class

File: Detectors/MUON/MCH/Raw/ElecMap/include/MCHRawElecMap/Mapper.h (hunk 1 above)
Comment (aphecetche) [reply Detectors/MUON/MCH/Raw/ElecMap/include/MCHRawElecMap/Mapper.h]: I also wondered where to put it, but decided against Constants as the first implementation I had was quite long. Now it's smaller so could fit in an include file problably. Let's try (requires to move the corresponding test as well).

File: Detectors/MUON/MCH/Raw/ElecMap/src/ElectronicMapperDummy.cxx (hunk 2 above)
Comment (aphecetche) [reply Detectors/MUON/MCH/Raw/ElecMap/src/ElectronicMapperDummy.cxx:179]: not dramatic as the Dummy is supposed to disappear at some point, but I've added it for consistency. thanks.

File: Detectors/MUON/MCH/Constants/include/MCHConstants/DetectionElements.h [hunk 3]
```diff
@@ -34,6 +36,33 @@ inline bool isValidDetElemId(int deId)
 {
   return std::find(begin(deIdsForAllMCH), end(deIdsForAllMCH), deId) != deIdsForAllMCH.end();
 }
+
+/** deIndex returns an index (in the range 0..155) corresponding the deId. */
+std::optional<int> deId2DeIndex(int deId);
+
```
Comment (pillot) [inline Detectors/MUON/MCH/Constants/include/MCHConstants/DetectionElements.h]: you can remove this extra declaration.

File: Detectors/MUON/MCH/Raw/ElecMap/src/Mapper.cxx [hunk 4]
```diff
@@ -10,6 +10,9 @@
 // or submit itself to any jurisdiction.
 
 #include "MCHRawElecMap/Mapper.h"
+#include "MCHConstants/DetectionElements.h"
+#include <optional>
+#include <unordered_map>
```
Comment (pillot) [inline Detectors/MUON/MCH/Raw/ElecMap/src/Mapper.cxx]: I suppose these 3 includes are not needed anymore

File: Detectors/MUON/MCH/Raw/ElecMap/src/testElectronicMapper.cxx [hunk 5]
```diff
@@ -17,12 +17,14 @@
...
+
+#include "dslist.h"
+#include <array>
 #include <boost/mpl/list.hpp>
+#include <fmt/format.h>
 #include <gsl/span>
-#include <array>
-#include "dslist.h"
+#include <set>
+#include <unordered_map>
```
Comment (pillot) [inline Detectors/MUON/MCH/Raw/ElecMap/src/testElectronicMapper.cxx]: I suspect the include of unordered_map is not needed anymore

File: Detectors/MUON/MCH/Constants/include/MCHConstants/DetectionElements.h (hunk 3 above)
Comment (aphecetche) [reply Detectors/MUON/MCH/Constants/include/MCHConstants/DetectionElements.h]: done

File: Detectors/MUON/MCH/Raw/ElecMap/src/testElectronicMapper.cxx (hunk 5 above)
Comment (aphecetche) [reply Detectors/MUON/MCH/Raw/ElecMap/src/testElectronicMapper.cxx]: indeed

File: Detectors/MUON/MCH/Raw/ElecMap/src/Mapper.cxx (hunk 4 above)
Comment (aphecetche) [reply Detectors/MUON/MCH/Raw/ElecMap/src/Mapper.cxx]: indeed

**Review by pillot — APPROVED** [review]

**Pull request merged**